src/DocumentDb/DocumentDbDocumentAdapter.cs

```diff
@@ -0,0 +1,475 @@
+					}
+					catch (DocumentClientException e) when (e.StatusCode == HttpStatusCode.NotFound)
+					{
+						return default(ResourceResponse<Document>);
+					}
+				});
+		}
+
+
+		/// <summary>
+		/// Gets a document and converts to a POCO type.
+		/// </summary>
+		/// <param name="dbId">Database id.</param>
+		/// <param name="collectionId">Collection id.</param>
+		/// <param name="docId">Document id.</param>
+		/// <param name="requestOptions">Request options.</param>
+		/// <returns>POCO object deserialized by default default json deserialization settings by document db sdk.</returns>
+		public async Task<T> GetDocumentAsync<T>(
+			string dbId, string collectionId, string docId, RequestOptions requestOptions = null) where T : class
+		{
+			Code.ExpectsNotNullOrWhiteSpaceArgument(dbId, nameof(dbId), 0);
+			Code.ExpectsNotNullOrWhiteSpaceArgument(collectionId, nameof(collectionId), 0);
+			Code.ExpectsNotNullOrWhiteSpaceArgument(docId, nameof(docId), 0);
+
+			IDocumentClient client = await GetDocumentClientAsync();
+
+
+			return await DocumentDbAdapter.ExecuteAndLogAsync(0,
+				async () =>
+				{
+					try
+					{
+						return await client.ReadDocumentAsync<T>(UriFactory.CreateDocumentUri(dbId, collectionId, docId), requestOptions);
+					}
+					catch (DocumentClientException e) when (e.StatusCode == HttpStatusCode.NotFound)
+					{
+						return default(T);
+					}
+				});
+		}
+
+
+		/// <summary>
+		/// Gets a document and converts to a POCO using specified converter.
+		/// </summary>
+		/// <param name="dbId">Database id.</param>
+		/// <param name="collectionId">Collection id.</param>
+		/// <param name="docId">Document id.</param>
+		/// <param name="requestOptions">Request options.</param>
+		/// <param name="converter">Delegate to convert a document to a POCO object.</param>
+		/// <returns>POCO object converted from the retrieved Document.</returns>
+		public async Task<T> GetDocumentAsync<T>(
+			string dbId, string collectionId, string docId, Func<Document, T> converter, RequestOptions requestOptions = null)
+		{
+			Code.ExpectsNotNullOrWhiteSpaceArgument(dbId, nameof(dbId), 0);
+			Code.ExpectsNotNullOrWhiteSpaceArgument(collectionId, nameof(collectionId), 0);
+			Code.ExpectsNotNullOrWhiteSpaceArgument(docId, nameof(docId), 0);
+			Code.ExpectsArgument(converter, nameof(converter), 0);
+
+			IDocumentClient client = await GetDocumentClientAsync();
+
+			ResourceResponse<Document> response = await DocumentDbAdapter.ExecuteAndLogAsync(
+				0, () => client.ReadDocumentAsync(
+					UriFactory.CreateDocumentUri(dbId, collectionId, docId), requestOptions));
+
+			return converter(response.Resource);
+		}
+
+
+		/// <summary>
+		/// Gets all documents.
+		/// </summary>
+		/// <param name="dbId">Database id.</param>
+		/// <param name="collectionId">Collection id.</param>
+		/// <param name="feedOptions">Feed options.</param>
+		/// <param name="queryFunc">Optional query</param>
```

Comment:
I take it this query is there to filter the document set. If so, suggest being more verbose in the comment to make that clear.